### ISSUE TITLE
Make it simpler in `ExecutionDiagnosticEventListener.ResolverError` to distinguish between server and client errors

### ISSUE THREAD

**Issue by cmeeren** [ChilliCream/graphql-platform#7144]
### Product

Hot Chocolate

### Is your feature request related to a problem?

I want to log all server errors as errors, and all client errors at a lower severity level. However, `ExecutionDiagnosticEventListener.ResolverError` gives me no robust way to distinguish between them.

I originally tried to base this on whether the `IError` passed to `ResolverError` contains an exception, but this does then not log errors like `HC0018` "Cannot return null for non-nullable field" as errors. I can of course add this code to the check, but adding error codes as I randomly discover that something is not logged as an error is no way to go about this.

### The solution you'd like

Something as simple as `IError.IsServerError`.

**Comment by michaelstaib** [comment]
With client errors you really mean error that are caused by client inputs?

**Comment by cmeeren** [comment]
Yes, that is correct. Put another way, errors that according to normal HTTP semantics would give 4XX errors (and not 5XX errors). Or in yet other words: Errors where the client, not the server, is at fault.

**Comment by cmeeren** [comment]
Related: Error code `HC0018` is used both [when a required variable is missing](https://github.com/ChilliCream/graphql-platform/blob/a4fc9333e0cd2c87b71db28cb16a9ca8b7591a9a/src/HotChocolate/Core/test/Execution.Tests/Processing/__snapshots__/VariableCoercionHelperTests.Error_When_Value_Is_Null_On_Non_Null_Variable.snap#L4) (client error) and [when the server tries to return null for a non-nullable field](https://github.com/ChilliCream/graphql-platform/blob/a4fc9333e0cd2c87b71db28cb16a9ca8b7591a9a/src/HotChocolate/Fusion/test/Core.Tests/__snapshots__/ErrorTests.Accounts_Offline_Author_NonNull.md?plain=1#L23) (server error).

So currently, for this error code, I have to also check the message in order to know whether to log this as an error.
